libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c

```diff
@@ -389,6 +389,64 @@ Socket_t SOCKETS_Socket( int32_t lDomain,
 
 /*-----------------------------------------------------------*/
 
+int32_t SOCKETS_Bind( Socket_t xSocket,
```

Comment:
We have unit tests for the LWIP implementation here: https://github.com/aws/amazon-freertos/blob/master/libraries/abstractions/secure_sockets/utest/secure_sockets_utest.c

Would you be able to add unit tests for this function?

Reply:
Hi @aggarw13 ,

What is the expectation here just add one API similar to below with bind and call in test_xx functions, please let me know in how many test you would like to call bind 

static Socket_t create_normal_connection()
{
    Socket_t s = initSocket();

    connectSocket( s );
    return s;
}

Thanks
Balaji.

Reply:
It would be great if you can add unit tests to provide full line and branch coverage in the definition of the `SOCKETS_Bind` function being tested. Therefore, it would be good to have tests for invalid input parameters (as a  `test_SecureSockets_Bind_Invalid_Input` test), tests for different failure path from the `lwip` functions (i.e. failures from the mocked `lwip_setsockopt` and `lwip_bind` functions) and a test for the success path (where all the mocked lwip functions return success values).

Reply:
@aggarw13 

I have added unit test case for bind for invalid and a test for success path.
Unfortunately I am not able to build/verify the test coverage due to setup issues with Ubuntu (18.04) by following the URL  https://github.com/aws/amazon-freertos/blob/master/tests/unit_test/linux/README.md 

Thanks
Balaji.